### PR TITLE
fix!: Make a label to be mandatory for a TestStep

### PR DESCRIPTION
**Pull request by xaionaro**
ITS: https://github.com/facebookincubator/contest/issues/13

### CODE REVIEW

File: pkg/pluginregistry/bundles.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ func (r *PluginRegistry) NewTestStepBundle(testStepDescriptor test.TestStepDescr
 	}
 	label := testStepDescriptor.Label
 	if label == "" {
-		label = testStepDescriptor.Name
+		return nil, ErrStepLabelIsMandatory{testStepDescriptor}
```
Comment (insomniacslk) [inline pkg/pluginregistry/bundles.go]: Struct initialization without explicit field name should be avoided - I thought we had a linter check in place for that

File: tests/integ/jobmanager/common.go [hunk 2]
```diff
@@ -424,3 +424,23 @@ func (suite *TestJobManagerSuite) TestJobManagerJobCancellationFailure() {
...
 
 }
+
+func (suite *TestJobManagerSuite) TestTestStepNoLabel() {
+	go func() {
+		suite.jm.Start(suite.sigs)
+		close(suite.jobManagerCh)
+	}()
+
+	_, err := suite.startJob(jobDescriptorNoLabel)
```
Comment (marcoguerri) [inline tests/integ/jobmanager/common.go]: Since we are returning `ErrStepLabelIsMandatory`, can we assert error type here?

**Comment by marcoguerri** [comment]
Also, I think we should look into moving some error definitions that we have in `cerrors` into the packages they belong to and actually get rid of that package.

File: tests/integ/jobmanager/common.go (hunk 2 above)
Comment (xaionaro) [reply tests/integ/jobmanager/common.go]: Yeah, I was going to do so. But it (`errors.As`) was implemented only in Go1.13, while this package is checked by Go1.11 (or so) in CI. So I decided not to add an extra dependency for an assert only.

File: pkg/pluginregistry/bundles.go (hunk 1 above)
Comment (xaionaro) [reply pkg/pluginregistry/bundles.go]: Fixed.

File: tests/integ/jobmanager/common.go (hunk 2 above)
Comment (xaionaro) [reply tests/integ/jobmanager/common.go]: Fixed.

**Comment by xaionaro** [comment]
> Also, I think we should look into moving some error definitions that we have in cerrors into the packages they belong to and actually get rid of that package (which is the approach used here).

I did not notice this package. Should I move the new error there (while `cerror` still exists)?

**Comment by marcoguerri** [comment]
> I did not notice this package. Should I move the new error there (while cerror still exists)?

I would leave them as is, and instead get rid of `cerrors` (I'll open an issue)

File: tests/integ/jobmanager/common.go (hunk 2 above)
Comment (marcoguerri) [reply tests/integ/jobmanager/common.go]: Mh.. we should also fix the Travis configuration to use go 1.13, integration test are run in a go 1.13 container. Thank you for spotting this.

**Comment by xaionaro** [comment]
Please note that I checked verb `%v` to `%w` in one place. It will cause less neat error messages in Go<1.13. It will still compile and work, but still.

**Comment by marcoguerri** [comment]
Apologies, I also did not merge this. Could you please look into the merge conflict (trivial json descriptor...). Feel free to merge autonomously afterwards.

**Comment by xaionaro** [comment]
> Apologies, I also did not merge this. Could you please look into the merge conflict (trivial json descriptor...).

No problem, done :)

> Feel free to merge autonomously afterwards.

I don't have privileges in this repository/project.

**Comment by marcoguerri** [comment]
We should fix the privileges problem.

**Comment by xaionaro** [comment]
> We should fix the privileges problem

OK. I've just added myself to the team :)

**Comment by marcoguerri** [comment]
Time to go.

**Pull request merged**